File: 1038. 统计同成绩学生.py
import sys
N = int ( sys.stdin.readline() )
Score = map( int , sys.stdin.readline().split() ) 
K , Search = sys.stdin.readline().strip().split( maxsplit = 1 )
Data = [ 0 for i in range(101) ]
for i in Score :
    Data[ i ] += 1
flag = 0
Search = map( int , Search.split() )
for i in Search :
    if not flag :
        print( Data[ i ],end='')
        flag = 1
    else :
        print( '' , Data[ i ],end='')

File: 1038. 统计同成绩学生.c
#include <stdio.h>
int main( void )
{
        int N;
        scanf( "%d" , &N ) ;
        int Score[ 101 ] = { 0 } ;
        for( int i = 0 , tem ; i < N ; i++ )
        {
                scanf( "%d" , &tem ) ;
                Score[ tem ] ++ ;
        }
        scanf( "%d" , &N ) ;
        for( int i = 0 , tem , flag = 0 ; i < N ; i++ )
        {
                scanf( "%d" ,&tem ) ;
                if( !flag )
                {
                        printf( "%d" , Score[ tem ] ) ;
                        flag = 1 ;
                }
                else
                        printf(" %d" , Score[ tem ] ) ;
        }
        return 0 ;
}
//除去读入数据的方式不同，逻辑和运算过程与python代码一摸一样，但python的运行速度慢了接近6倍
